packages/editor/src/components/post-preview-button/index.js

```diff
@@ -211,15 +211,19 @@ export default compose( [
 		} = select( 'core/editor' );
 		const {
 			getPostType,
+			hasFetchedAutosaves,
 		} = select( 'core' );
 
 		const previewLink = getEditedPostPreviewLink();
-		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+		const postTypeName = getEditedPostAttribute( 'type' );
+		const postType = getPostType( postTypeName );
+		const postId = getCurrentPostId();
+
 		return {
 			postId: getCurrentPostId(),
 			currentPostLink: getCurrentPostAttribute( 'link' ),
 			previewLink: forcePreviewLink !== undefined ? forcePreviewLink : previewLink,
-			isSaveable: isEditedPostSaveable(),
+			isEnabled: isEditedPostSaveable() && hasFetchedAutosaves( postTypeName, postId ),
```

Comment:
I'm struggling to understand what implication the availability of fetched autosaves has on whether the post preview button should be disabled. At the least, could we include an inline code comment somewhere which explains this?

Reply:
The check is to avoid an autosave attempt being made (via the preview button) when the user already has an autosave that the client hasn't fetched. That would cause a situation where the client and the server disagree, and the server can throw an error (basically the same error that this PR is fixing).

On reflection, I don't think the right strategy. I've pushed a change instead to `isEditedPostAutosaveable` so that it returns false if autosaves have not been fetched:
https://github.com/WordPress/gutenberg/blob/542b4ba844de091636625f1216cd243153749328/packages/editor/src/store/selectors.js#L539-L543

Now, if autosaves have not been fetched, clicking the preview button will cause it to fallback to the existing preview link or post link and it won't attempt an autosave during that process.